esm_master/compile_info.py

```diff
@@ -498,6 +516,12 @@ def list_all_packages(self, vcs, general):
         for kind in self.model_kinds:
             for model in config[kind]:
                 version = None
+
+                # if the [kind][model] is not an iterable like list or dict
+                # then continue with the next item
+                if not isinstance(config[kind][model], Iterable):
+                    continue
+
```

Comment:
I think if a `model` is not iterable, we want to catch the error, therefore, I would remove this lines if possible.

Reply:
Yes, this is also what I thought. `kind` becomes `use_database` and `model` becomes `False`. I think when Dirk first formulated these config files, he did not consider the possiblity to add other type of options here. I think for the rest of the options this is fine but for scalar / Boolean options this can be a problem.

Reply:
The `kind` logic is very important to `esm_master` and I am concern about errors that can be hidden by these lines. I would actually remove this lines and instead of loading the `general.yaml` info where you are loading it, I would load that file after the following lines, so that we make sure it makes it into the `config["general"]`:
https://github.com/esm-tools/esm_master/blob/21c024f68ff52e27f10773cf90df604f5e96c462/esm_master/general_stuff.py#L74-L120